test_jpeg/test/test_rle.py

```diff
@@ -0,0 +1,184 @@
+        yield clock.posedge
+
+    print ("runlength %d size %d amplitude %d" % (
+        bufferdatabus.runlength, bufferdatabus.size, bufferdatabus.amplitude))
+
+    bufferdatabus.read_enable.next = False
+    yield clock.posedge
+
+@block
+def test():
+
+    clock = Signal(bool(0))
+    reset = ResetSignal(0, active=1, async=True)
+
+    indatastream = InDataStream()
+    bufferdatabus = BufferDataBus()
+    rleconfig = RLEConfig()
+
+    jpgv1 = rletop(reset, clock, indatastream, bufferdatabus, rleconfig)
+   
+
+    @instance
+    def tbclock():
+        clock.next=0
+        while True:
+            yield delay(10)
+            clock.next = not clock
+
+
+    @instance
+    def tbsim():
+        """ reset signal """
+        yield reset_on_start(clock, reset)
+
+        bufferdatabus.buffer_sel.next = False
+        yield clock.posedge
+        yield write_block(
+            clock, red_pixels_1,
+            indatastream,
+            bufferdatabus,
+            rleconfig,  red
+            )
+        yield clock.posedge
+        print ("=====")
+
+        yield read_block(True, bufferdatabus, clock)
+
+        yield write_block(
+            clock, red_pixels_2,
+            indatastream,
+            bufferdatabus,
+            rleconfig,  red
+            )
+        yield clock.posedge
+
+        print ("=====")
+
+        yield read_block(False, bufferdatabus, clock)
+
+        yield write_block(
+            clock, green_pixels_1,
+            indatastream,
+            bufferdatabus,
+            rleconfig, green
+            )
+        yield clock.posedge
+        print ("=======")
+
+
+        yield read_block(True, bufferdatabus, clock)
+
+        yield write_block(
+            clock, green_pixels_2,
+            indatastream,
+            bufferdatabus,
+            rleconfig, green
+            )
+        yield clock.posedge
+        print ("=======")
+
+
+        yield read_block(False, bufferdatabus, clock)
+
+        yield write_block(
+            clock, blue_pixels_1,
+            indatastream,
+            bufferdatabus,
+            rleconfig, blue
+            )
+        yield clock.posedge
+        print ("=======")
+
+
+        yield read_block(True, bufferdatabus, clock)
+
+        yield write_block(
+            clock,  blue_pixels_2,
+            indatastream,
+            bufferdatabus,
+            rleconfig, blue
+            )
+        yield clock.posedge
+        print ("=======")
+
+        yield read_block(False, bufferdatabus, clock)
+
+        print ("===============")
+
+        yield clock.posedge
+        rleconfig.sof.next = True
+        yield clock.posedge
+
+        raise StopSimulation        
+    return  tbsim, tbclock, jpgv1
+
+
+
+def conversion_bench():
+
+    instance = test()
+    instance.config_sim(trace = True)
+    instance.run_sim()
+    #verify.simulator='iverilog'
+    #assert testbench2().verify_convert() == 0
+
+if __name__ == '__main__':
+    conversion_bench() 
```

Comment:
The tests look good but they need to be organized differently so the py.test test runner will execute the code, see pytest.org.